airflow/www/static/js/utils/index.ts

```diff
@@ -49,6 +48,44 @@ const finalStatesMap = () =>
     ["no_status", 0],
   ]);
 
+interface GroupSummaryProps {
+  group: Task;
+  runId?: string;
+  mappedStates: TaskInstance["mappedStates"];
+}
+
+const getGroupAndMapSummary = ({
+  group,
+  runId,
+  mappedStates,
+}: GroupSummaryProps) => {
+  let totalTasks = 0;
+  const childTaskMap = finalStatesMap();
+  if (!!group.children && !group.isMapped) {
+    group.children?.forEach((child) => {
+      const taskInstance = child.instances.find((ti) => ti.runId === runId);
+      if (taskInstance) {
+        const stateKey =
+          taskInstance.state == null ? "no_status" : taskInstance.state;
+        if (childTaskMap.has(stateKey)) {
+          childTaskMap.set(stateKey, (childTaskMap.get(stateKey) || 0) + 1);
+        }
+      }
+    });
+  } else if (group.isMapped && mappedStates) {
+    Object.keys(mappedStates).forEach((stateKey) => {
```

Comment:
Nit: If you need access to keys and values, Object.entries can be a little easier. It has less support than Object.keys but babel should be able to transform that. (We might want to check, I don't see the browserlist target in our package.json)